KEYCLOAK-12937 adding OSGi metadata to module manifest

Neutius opened the pull request:
The Keycloak Installed Application adapter module currently has no OSGi-metadata in its MANIFEST.MF file.

Several other modules do have OSGi-metadata their respective manifests, such as Keycloak Core, Keycloak Adapter Core, and Keycloak Authz Client. I looked at those modules for examples, to ensure consistency between the manifests of different modules.

I compared the results of different modules, and I'm satisfied with the results.

I have tested this for an Eclipse RCP application, and it works without issue. Some of this module's dependencies are not readily available as OSGi bundles - my company OSGi-fies libraries for internal use.

Neutius committed:
KEYCLOAK-12937 adding OSGi metadata to module MANIFEST.MF for Keycloak Installed

Neutius commented:
@hmlnarik Would you be able to review this pull request?

For background information, see:
- the Jira story here: https://issues.redhat.com/browse/KEYCLOAK-12937 
- the discussion on the mailing list here: https://groups.google.com/forum/#!topic/keycloak-dev/HmG3wfINsgw

hmlnarik reviewed (changes requested):
@Neutius Thanks for the PR and apologies for delay.

Could you please update version limits for all the dependencies? For the `org.keycloak` dependencies, the version should be like [this](https://github.com/keycloak/keycloak/blob/8.0.1/adapters/oidc/adapter-core/pom.xml#L41), org.jboss.resteasy.client.jaxrs should have at least major version, javax.* need bounds to match the correct specification (like [here](https://github.com/keycloak/keycloak/blob/8.0.1/adapters/oidc/undertow/pom.xml#L42))

Neutius commented:
@hmlnarik Thanks for your reply, and your interest in this Pull Request.

It's been a while, but I'm sure setting version ranges like that is redundant. I added a generated Manifest file to the Jira story, in which the version ranges are present. Apache Felix looks at the pom files and deduces the version ranges, usually correctly ime. [See the Jira issue](https://issues.redhat.com/browse/KEYCLOAK-12937)

I rebuild the project locally, and added the newer manifest to the story as well. Versioning looks like this:

`Export-Package: org.keycloak.adapters.installed;uses:="javax.crypto,org.
 jboss.resteasy.client.jaxrs,org.keycloak.jose.jwe,org.keycloak.represen
 tations.idm,org.keycloak.representations.adapters.config,javax.ws.rs.cl
 ient,javax.crypto.spec,javax.ws.rs.core,org.keycloak.representations,or
 g.keycloak,org.keycloak.common.util,org.keycloak.adapters,org.keycloak.
 util,org.keycloak.common,org.keycloak.adapters.rotation";version="9.0.0
 .SNAPSHOT"
Import-Package: javax.crypto,javax.crypto.spec,javax.ws.rs.client;versio
 n="[2.0,3)",javax.ws.rs.core;version="[2.0,3)",org.jboss.resteasy.clien
 t.jaxrs,org.keycloak;version="[9.0,10)",org.keycloak.adapters;version="
 [9.0,10)",org.keycloak.adapters.rotation;version="[9.0,10)",org.keycloa
 k.common;version="[9.0,10)",org.keycloak.common.util;version="[9.0,10)"
 ,org.keycloak.jose.jwe;version="[9.0,10)",org.keycloak.representations;
 version="[9.0,10)",org.keycloak.representations.adapters.config;version
 ="[9.0,10)",org.keycloak.representations.idm;version="[9.0,10)",org.key
 cloak.util;version="[9.0,10)"`

This looks good, as far as I can tell. If you still prefer setting version ranges explicitly to keep more control, I'll be happy to do it, but I feel that would lead to unnecessary maintenance work.

hmlnarik commented:
Thank you for your inquiry. Closing this PR. Feel free to reopen with the requested changes.
